spacy/pipeline/morphologizer.pyx

```diff
@@ -217,14 +226,34 @@ class Morphologizer(Tagger):
             docs = [docs]
         cdef Doc doc
         cdef Vocab vocab = self.vocab
+        overwrite = self.cfg.get("overwrite", False)
+        extend = self.cfg.get("extend", False)
         for i, doc in enumerate(docs):
             doc_tag_ids = batch_tag_ids[i]
             if hasattr(doc_tag_ids, "get"):
                 doc_tag_ids = doc_tag_ids.get()
             for j, tag_id in enumerate(doc_tag_ids):
                 morph = self.labels[tag_id]
-                doc.c[j].morph = self.vocab.morphology.add(self.cfg["labels_morph"].get(morph, 0))
-                doc.c[j].pos = self.cfg["labels_pos"].get(morph, 0)
+                # set morph
+                if doc.c[j].morph == 0 or overwrite or extend:
```

Comment:
I understand what's happening with `morph == 0` but it took me a second to understand it and I still think it looks weird (though it's also the best way to check this). Maybe there should be a comment here?

Reply:
This is how we always check for unset values in `set_annotations`.